magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,374 @@
+"""
+This scripts facilitates the usage of the script
+"lst1_magic_event_coincidence.py". This script is
+more like a "manager" that organizes the analysis
+process by:
+1) Creating the bash scripts for looking for
+coincidence events between MAGIC and LST in each
+night.
+2) Creating the subdirectories for the coincident
+event files.
+
+Usage:
+$ python coincident_events.py (-c config_file.yaml)
+"""
+import argparse
+import glob
+import logging
+import os
+from datetime import date as dtdt
+from datetime import timedelta
+from pathlib import Path
+
+import numpy as np
+import yaml
+
+from magicctapipe import __version__
+
+__all__ = ["configfile_coincidence", "linking_bash_lst"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configfile_coincidence(ids, target_dir):
+
+    """
+    This function creates the configuration file needed for the event coincidence step
+
+    Parameters
+    ----------
+    ids : list
+        List of telescope IDs
+    target_dir : str
+        Path to the working directory
+    """
+
+    lines = [
+        f"mc_tel_ids:\n    LST-1: {ids[0]}\n    LST-2: {ids[1]}\n    LST-3: {ids[2]}\n    LST-4: {ids[3]}\n    MAGIC-I: {ids[4]}\n    MAGIC-II: {ids[5]}\n\n",
+        'event_coincidence:\n    timestamp_type_lst: "dragon_time"  # select "dragon_time", "tib_time" or "ucts_time"\n    pre_offset_search: true\n    n_pre_offset_search_events: 100\n    window_half_width: "300 ns"\n',
+        '    time_offset:\n        start: "-10 us"\n        stop: "0 us"\n',
```

Comment:
I do not like this approach much, because it hardcodes the settings inside the script, and if we change them in the general config file they are not propagated. How about instead you pass the (path to) general config file instead, and extract from it mc_tel_ids:  and event_coincidence: ?

Reply:
Yes for sure. We already tried to fix this in the setting_up script: we can use the same method also here and in stereo, if the method is fine for you

Reply:
I agree with Julian here, the general config has all the information needed already.

Reply:
Fixed (?)